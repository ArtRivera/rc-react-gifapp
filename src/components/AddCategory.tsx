import React,{ChangeEvent, FormEvent, useState,useEffect} from 'react';

interface Props{
    setCategory: any
}

const AddCategory = ({setCategory}: Props) => {

    const [inputValue, setInputValue] = useState('');

    useEffect(() => {
        console.log(inputValue);
    }, [inputValue])

    const handleInputChange = (e: ChangeEvent<HTMLInputElement>) =>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e: FormEvent<HTMLFormElement>) =>{
        e.preventDefault();
        
        if(inputValue.trim().length > 2){
            setCategory((cats:any) => [inputValue,...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
            />
        </form>
    )
}

export default AddCategory
