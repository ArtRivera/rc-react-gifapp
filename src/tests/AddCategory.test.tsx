import './setupTests';
import { shallow, ShallowWrapper } from 'enzyme'
import AddCategory from '../components/AddCategory'

describe('Prueba de AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper: ShallowWrapper;

    beforeEach(() =>{
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory  setCategory={setCategories}/>);
    })

    test('should mostrarse bien', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should csmbiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change',{ target: { value}});
    });

    test('No debe de postear el onSubmit', () => {
        const form = wrapper.find('form');
        form.simulate('submit',{
             preventDefault(){}
    });
    expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('should llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Testyyyy'
        wrapper.find('input').simulate('change', {
            target: {value} 
         });
        wrapper.find('form').simulate('submit',{ preventDefault(){}});

        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');

    })
    
    
})
