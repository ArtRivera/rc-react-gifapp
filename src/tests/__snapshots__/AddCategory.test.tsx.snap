// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de AddCategory should mostrarse bien 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
